### DISCUSSION TITLE
Typescript issue when publishing library that takes a `PublicClient` as an argument, if we don't cast the `chain` argument as `Chain`

### DISCUSSION THREAD

**Discussion by oveddan** [wevm/viem#2282] [General]
In our library `@zoralabs/protocol-sdk` we publish some client creation methods that [take a `PublicClient` as an argument](https://github.com/ourzora/zora-protocol/blob/main/packages/protocol-sdk/src/create/1155-create-helper.ts#L246).

When trying to integrate with this package externally, we get typescript errors when passing a `PublicClient` as an argument.  Here is some example code:

```ts
import { createPublicClient, http } from 'viem'
import { zora } from "viem/chains";
import { create1155CreatorClient } from "@zoralabs/protocol-sdk";

const publicClient = createPublicClient({
  chain: zora,
  transport: http(),
});

const client = create1155CreatorClient({
  publicClient
});
```

This gives us typescript error: "The types returned by 'getBlock(...)' are incompatible between these types."

See this error in [the typescript playground](https://www.typescriptlang.org/play/?ts=5.4.5#code/JYWwDg9gTgLgBAbzgYygUwIYzQBQK4BGANsMgMIloB2MANHABYwxj1kMbBVwC+cAZlAgg4AcgBuwNCFEAoUJFiIAXtAx9BwuACJJ0gPTIOXAM7aA3PPDR4SVJmwBGRwFYXZdFmgUpNXgKERbQABVSgMIgwCE30wIRgIZAgiAFoTABMAawtZWSSqE3gwQhJySj8AXhRPbHxiUh9qGAAKBFk4FGMqAC44MIxadrgYcILFGF6mFmaASkGeGcs8iAL4ZHL4KvssNGc3DwdvDdah4vqy3xhZBcsgA)

However, if we append `as Chain` to the chain argument, the error goes away.

```ts
import { createPublicClient, http, Chain } from 'viem'
import { zora } from "viem/chains";
import { create1155CreatorClient } from "@zoralabs/protocol-sdk";

const publicClient = createPublicClient({
  chain: zora as Chain,
  transport: http(),
});

const client = create1155CreatorClient({
  publicClient
});
```

Is there a way types can be better defined without requiring us to use this `as Chain` argument?

**Comment by jxom** [comment]
Yes. You will need to properly hook up the generics in the library code. Here you go: 

https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgYygUwIYzQBQK4BGANsMgMIloB2MANHABYwxj1kMbBVuU30AqUDFQDOkWHAC+cAGZQIIOACIAbsDQglAbgBQocfCQAvaBimz5i1epAB6ZBy4jtOnQBM0yIhnSy8VZBhgCCoUdCw0AEZIgFYYsnCYaAp1GgAeB05QtAAPbCo3ETh2LIA+AAoEHTg4MEIScl46HUkALkRq2vrSFOoYdt70wWExaDoURypS3UkASnaVCGA3XR1kEJF4OuIeprgAXjDMbHwdxtSYSs7MrnaTIVpOmCFRA3amFnLZx7nV9dF4F4LgcjhFonEEsdkk0rjVtg1Bs1fjogA

**Reply by oveddan** [reply]
thanks that seemed to fix it!
